src/interface/src/app/top-bar/top-bar.component.ts

```diff
@@ -18,52 +12,42 @@ import { FeatureService } from '../features/feature.service';
   templateUrl: './top-bar.component.html',
   styleUrls: ['./top-bar.component.scss'],
 })
-export class TopBarComponent implements OnInit, OnDestroy {
+export class TopBarComponent implements OnInit {
   @Output()
   toggleEvent = new EventEmitter<Event>();
 
-  // display name is first name and then username
-  displayName: string = '';
-  loggedIn = false;
-
   readonly color = 'primary';
 
-  private readonly destroy$ = new Subject<void>();
-
   loggedIn$ = this.authService.isLoggedIn$;
 
+  displayName$ = this.loggedIn$.pipe(
+    switchMap((loggedIn) => this.authService.loggedInUser$),
+    map((user) => {
+      if (user === undefined) {
+        return;
+      }
+
+      if (user) {
+        return user.firstName || user.username;
+      } else {
+        return 'Guest';
+      }
+    })
+  );
+
+  initial$ = this.displayName$.pipe(
+    map((displayName) => displayName?.substring(0, 1))
+  );
+
   constructor(
     private authService: AuthService,
     private dialog: MatDialog,
-    private router: Router,
-    private featureService: FeatureService,
-    private route: ActivatedRoute
+    private router: Router
   ) {}
 
   ngOnInit(): void {
     this.router.routeReuseStrategy.shouldReuseRoute = () => false;
-    this.authService.loggedInUser$
-      .pipe(takeUntil(this.destroy$))
-      .subscribe((user) => {
-        if (user) {
-          this.displayName = user.firstName
```

Comment:
This was unnecessarily complicated. Our sign up form has name required so I don't think we should handle the scenario of not having name. Just in case I'm still defaulting to username if name is not found (see line 39)